src/main/java/corgi/storage/Storage.java

```
@@ -0,0 +1,107 @@
+                boolean success = file.createNewFile();
+                if (!success) {
+                    System.err.println("An error occurred when creating file: " + file.getAbsolutePath());
+                } else {
+                    System.out.println("No previous data file found. Created a new data file at: "
+                            + file.getAbsolutePath());
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("An error occurred when creating file: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Saves a list of storable items to a file.
+     *
+     * @param list The list of storable items to be saved.
+     */
+    public void save(StorableList<T> list) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(this.filePath))) {
+            writer.write(list.getStorableString());
+        } catch (IOException e) {
+            System.err.println("An error occurred when saving data: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Loads a list of items from a file and returns them as a List.
+     *
+     * @return A List containing the loaded items.
+     */
+    public List<T> load() {
+        List<T> items = new ArrayList<>();
+
+        int skipped = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(this.filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                try {
+                    T item = parser.parse(line);
+                    if (item != null) {
+                        items.add(item);
+                    }
+                } catch (ParsingException e) {
+                    skipped += 1;
+                    System.err.println("A parsing error occurred when loading data: " + e.getMessage()
+                            + "\nCorrupted data: " + line + " (Skipped)");
+                }
+            }
+        } catch (IOException e) {
```

No empty catch blocks! Nice!! 😄